drivers/gpio/gpio_ifx_cat1.c

```diff
@@ -202,27 +165,28 @@ static uint32_t gpio_cat1_get_pending_int(const struct device *dev)
 	return GPIO_PRT_INTR_MASKED(base);
 }
 
-static void gpio_event_callback(void *callback_arg, cyhal_gpio_event_t event)
+static void gpio_isr_handler(const struct device *dev)
 {
-	ARG_UNUSED(event);
-	uint32_t port_num = CYHAL_GET_PORT((uint32_t) callback_arg);
-	uint32_t pin_num = CYHAL_GET_PIN((uint32_t) callback_arg);
-	const struct device *dev = port_dev_obj[port_num];
+	const struct gpio_cat1_config *const cfg = dev->config;
+	GPIO_PRT_Type *const base = cfg->regs;
+	uint32_t pins = GPIO_PRT_INTR(base);
```

Comment:
you should use GPIO_PRT_INTR_MASKED instead GPIO_PRT_INTR, to check masked interrupts.

Reply:
updated..